2024-07-11 12:51:33.870591: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:51:33.870771: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:51:43,374 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:51:43,467 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:51:43,467 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:51:43,468 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:51:44,484 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:51:44,484 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:44,502 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.47}
INFO flwr 2024-07-11 12:51:44,502 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:51:45,169 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:51:45,204 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:51:45,206 | fedavgm.py:198 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,215 | server.py:131 | fit progress: (1, 0.34898479123648696, {'accuracy': 0.8366666666666667}, 0.7121137999999991)
DEBUG flwr 2024-07-11 12:51:45,215 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:51:45,257 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:51:45,257 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:51:45,257 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:51:45,308 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,319 | server.py:131 | fit progress: (2, 0.35174640748944724, {'accuracy': 0.8166666666666667}, 0.8168965000000004)
DEBUG flwr 2024-07-11 12:51:45,319 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:51:45,353 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:51:45,353 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:51:45,407 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,418 | server.py:131 | fit progress: (3, 0.3517440300007892, {'accuracy': 0.8166666666666667}, 0.9156704999999992)
DEBUG flwr 2024-07-11 12:51:45,418 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:51:45,474 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:51:45,474 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:51:45,537 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,549 | server.py:131 | fit progress: (4, 0.3552968083376014, {'accuracy': 0.8133333333333334}, 1.0471445999999993)
DEBUG flwr 2024-07-11 12:51:45,549 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:51:45,588 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:51:45,589 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:51:45,685 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,699 | server.py:131 | fit progress: (5, 0.3552909572620848, {'accuracy': 0.8133333333333334}, 1.1964232999999993)
DEBUG flwr 2024-07-11 12:51:45,700 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:45,741 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:51:45,741 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:45,799 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,810 | server.py:131 | fit progress: (6, 0.3545442919425989, {'accuracy': 0.8166666666666667}, 1.3084945000000001)
DEBUG flwr 2024-07-11 12:51:45,811 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:45,844 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:51:45,845 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:45,896 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:45,908 | server.py:131 | fit progress: (7, 0.354565239945104, {'accuracy': 0.8166666666666667}, 1.4055351999999992)
DEBUG flwr 2024-07-11 12:51:45,908 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:45,941 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:51:45,941 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:45,990 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:46,001 | server.py:131 | fit progress: (8, 0.3545520661751949, {'accuracy': 0.8166666666666667}, 1.4988610999999992)
DEBUG flwr 2024-07-11 12:51:46,001 | server.py:178 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:46,037 | server.py:192 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:51:46,037 | server.py:227 | fit_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:51:46,087 | server.py:241 | fit_round 9 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:46,100 | server.py:131 | fit progress: (9, 0.3545541205013387, {'accuracy': 0.8166666666666667}, 1.5976707999999995)
DEBUG flwr 2024-07-11 12:51:46,100 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:51:46,138 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:51:46,139 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:51:46,183 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:51:46,196 | server.py:131 | fit progress: (10, 0.35421422764671706, {'accuracy': 0.8166666666666667}, 1.6945724999999996)
DEBUG flwr 2024-07-11 12:51:46,197 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:51:46,229 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:51:46,229 | server.py:153 | FL finished in 1.7275454000000003
INFO flwr 2024-07-11 12:51:46,230 | app.py:225 | app_fit: losses_distributed [(1, 0.4125524200499058), (2, 0.4079054370522499), (3, 0.4109904250273338), (4, 0.41033364030031055), (5, 0.40788302031056634), (6, 0.4078028561740086), (7, 0.4078028828933321), (8, 0.4078029599683038), (9, 0.4076142404760633), (10, 0.40757542635713306)]
INFO flwr 2024-07-11 12:51:46,230 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:51:46,230 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:51:46,230 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.34898479123648696), (2, 0.35174640748944724), (3, 0.3517440300007892), (4, 0.3552968083376014), (5, 0.3552909572620848), (6, 0.3545442919425989), (7, 0.354565239945104), (8, 0.3545520661751949), (9, 0.3545541205013387), (10, 0.35421422764671706)]
INFO flwr 2024-07-11 12:51:46,230 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.47), (1, 0.8366666666666667), (2, 0.8166666666666667), (3, 0.8166666666666667), (4, 0.8133333333333334), (5, 0.8133333333333334), (6, 0.8166666666666667), (7, 0.8166666666666667), (8, 0.8166666666666667), (9, 0.8166666666666667), (10, 0.8166666666666667)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[-2.13189781,  1.01225179,  0.89503321,  0.56056078,  0.04155803,
         0.17214299,  0.07850318, -0.01774942]]), array([-0.22575837])]
[array([[-2.01599805,  1.02127524,  0.84882584,  0.4461299 ,  0.1887417 ,
         0.11053262, -0.02089052, -0.01805709]]), array([-0.1156154])]
[array([[-2.01612872,  1.02129619,  0.84892241,  0.44587521,  0.18879527,
         0.11050052, -0.02078243, -0.01793856]]), array([-0.11564525])]
[array([[-2.00253012e+00,  1.03474288e+00,  7.98986851e-01,
         4.33356810e-01,  1.61878569e-01,  1.12237912e-01,
        -7.57464983e-02, -1.63070251e-03]]), array([0.03089102])]
[array([[-2.00258866e+00,  1.03483320e+00,  7.99041056e-01,
         4.33620729e-01,  1.61860657e-01,  1.12414106e-01,
        -7.58240944e-02, -1.64823487e-03]]), array([0.03043203])]
[array([[-2.00355017,  1.03018549,  0.8269822 ,  0.41814934,  0.15269501,
         0.11326657, -0.06016723,  0.00224001]]), array([0.01234034])]
[array([[-2.00345094e+00,  1.03021635e+00,  8.26747805e-01,
         4.18090395e-01,  1.52633451e-01,  1.13500023e-01,
        -5.99556490e-02,  1.98068515e-03]]), array([0.01247894])]
[array([[-2.00349485,  1.03011696,  0.82693457,  0.4180757 ,  0.15263465,
         0.11335671, -0.06001613,  0.00215129]]), array([0.01246231])]
[array([[-2.00348965,  1.03012997,  0.82691111,  0.41807427,  0.15263668,
         0.11338058, -0.05999997,  0.00211927]]), array([0.01246204])]
[array([[-2.02017377,  1.01973644,  0.81860029,  0.41211169,  0.16029648,
         0.10341755, -0.0600709 ,  0.01531493]]), array([0.04371901])]
>>>>>>>>>>>>>run time:
0.53125s
